src/eureka/S5_lightcurve_fitting/differentiable_models/StarryModel.py

```diff
@@ -323,9 +399,81 @@ def update(self, newparams, **kwargs):
+                spotcon = np.zeros((self.nchannel_fitted, nspots))
+                starrot = np.zeros((self.nchannel_fitted))
+                starinc = np.zeros((self.nchannel_fitted))
+
+                # channel ID
+                if chan == 0:
+                    channel_id = ''
+                else:
+                    channel_id = f'_{chan}'
+
+                for n in range(nspots):
+                    # read radii
+                    item0 = 'spotrad' + str(n)
+                    if self.parameters.dict[item0][1] == 'free':
+                        item0 += channel_id
+                    value = self.parameters.dict[item0][0]
+                    spotrad[chan, n] = value
+                    # read latitudes
+                    item0 = 'spotlat' + str(n)
+                    if self.parameters.dict[item0][1] == 'free':
+                        item0 += channel_id
+                    value = self.parameters.dict[item0][0]
+                    spotlat[chan, n] = value
+                    # read longitudes
+                    item0 = 'spotlon' + str(n)
+                    if self.parameters.dict[item0][1] == 'free':
+                        item0 += channel_id
+                    value = self.parameters.dict[item0][0]
+                    spotlon[chan, n] = value
+                    # read contrasts 
+                    item0 = 'spotcon' + str(n)
+                    if self.parameters.dict[item0][1] == 'free':
+                        item0 += channel_id
+                    value = self.parameters.dict[item0][0]
+                    spotcon[chan, n] = value
+                # read stellar inclination
+                item0 = 'spotstari'
+                if self.parameters.dict[item0][1] == 'free':
+                    item0 += channel_id
+                value = self.parameters.dict[item0][0]
+                starinc[chan] = value
+                # read number of points
+                starres = self.parameters.dict['spotnpts'][0]
+                # read stellar rotation (if provided)
+                if 'spotrot' in self.parameters.dict.keys():
+                    starrot[chan] = self.parameters.dict['spotrot'][0]
+
+                # Initialize map object and add spots
+                map = starry.Map(ydeg=starres, udeg=self.udeg)
+                for si in range(nspots):
+                    map.spot(contrast=spotcon[chan, si], 
+                             radius=spotrad[chan, si],
+                             lat=spotlat[chan, si], 
+                             lon=spotlon[chan, si])
+
+                # Initialize star object
+                star = starry.Primary(map, m=0, r=pl_params.Rs, 
+                                      prot=starrot[chan])
+            else:           
+                # Initialize star object
+                star = starry.Primary(starry.Map(udeg=self.udeg),
+                                      m=0, r=pl_params.Rs)
```

Comment:
Almost all of my comments for the `setup` function above also apply here (except for the small number where I'd explicitly said they don't apply here)